### DISCUSSION TITLE
WiX v5: Add / Remove Local Users / Groups

### DISCUSSION THREAD

**Discussion by bevanweiss** [wixtoolset/issues#8562] [Questions] [answered]
Hi all,
Just wondering if there is already a pre-built Custom Action that I'm missing that would allow for creation / removal of Local Users / Groups during installation?

I can see there is a wixllib that appears to have been targetting v3
https://github.com/dblock/msiext/tree/master/src/WixExtensions/UserPrivilegesExtension

Is there a link to doco anywhere around a workflow to add a new extension into the current WiX v5 utils?
It seems like creating local users / user groups (and the memberships) is something similar to 'PermissionEx'

**Comment by robmen** [comment, marked as answer]
[util:User element](https://wixtoolset.org/docs/schema/util/user)?

**Reply by bevanweiss** [reply]
I’m obviously going blind. I’m sure I looked through all the options there.

Thank you greatly :)

**Reply by bevanweiss** [reply]
Ahh, I do see one glitch..
The creation of a local group, util:Group exists, but can’t create a group.

Would there be an objection to me adding the creation capability to this item?
So it could operate similar to util:User to create a user also..

**Reply by OlegSirenko** [reply]
Hi! I currently working on the creating local groups. Now I managed to create local groups with deferred custom actions. But is it possible to rebuild the original [UserPrivilegesExtension](https://github.com/dblock/msiext/tree/master/src/WixExtensions/UserPrivilegesExtension) as an extension for Wix v5 for using it nativly inside the .wxs file as a namespace?

**Reply by bevanweiss** [reply]
> But is it possible to rebuild the original [UserPrivilegesExtension](https://github.com/dblock/msiext/tree/master/src/WixExtensions/UserPrivilegesExtension) as an extension for Wix v5 for using it nativly inside the .wxs file as a namespace?

You'd need to update the UserPrivilegesExtension code to reference the WiX v5 interface points, instead of the v3 references it currently has.

**Reply by OlegSirenko** [reply]
Maybe there are some guideline how to make this? I tried to do this several times, but getting the error about Wix Factory...

**Reply by bevanweiss** [reply]
Start with an existing working Wix5 extension.

1. Copy it
2. rename it
3. Build it to ensure it works
4. then change the logic to suit your requirements.

Do NOT deviate from the ordering listed above.
